WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -695,9 +711,17 @@ public void onAuthenticationChanged(OnAuthenticationChanged event) {
             return;
         }
 
-        if (mAccountStore.hasAccessToken() && hasMagicLinkLoginIntent()) {
-            AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_MAGIC_LINK_SUCCEEDED);
-            ActivityLauncher.showLoginEpilogue(this, true, getIntent().getIntegerArrayListExtra(ARG_OLD_SITES_IDS));
+        if (mAccountStore.hasAccessToken()) {
+            // Check signup first since login and signup have the same action and host,
+            // but signup has an extra parameter.
+            if (hasMagicLinkSignupIntent()) {
```

Comment:
`hasMagicLinkSignupIntent()` seems to only check for the extra `new_user` query param without checking for the rest of the magiclink format so, maybe we can include it _inside_ the `hasMagicLinkLoginIntent()` conditional below to be more secure?

Reply:
I updated the logic in c1897d7d83bd8d5a35693c07c893321bafadedfc.